src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -159,6 +173,424 @@ struct kds_cu_info
+      {"Stream_Switch_Slave_East_2_Slot3", 0x0003F37C},
+      {"Stream_Switch_Slave_East_3_Slot0", 0x0003F380},
+      {"Stream_Switch_Slave_East_3_Slot1", 0x0003F384},
+      {"Stream_Switch_Slave_East_3_Slot2", 0x0003F388},
+      {"Stream_Switch_Slave_East_3_Slot3", 0x0003F38C},
+      {"Stream_Switch_Slave_ME_Trace_Slot0", 0x0003F390},
+      {"Stream_Switch_Slave_ME_Trace_Slot1", 0x0003F394},
+      {"Stream_Switch_Slave_ME_Trace_Slot2", 0x0003F398},
+      {"Stream_Switch_Slave_ME_Trace_Slot3", 0x0003F39C},
+      {"Stream_Switch_Slave_Mem_Trace_Slot0", 0x0003F3A0},
+      {"Stream_Switch_Slave_Mem_Trace_Slot1", 0x0003F3A4},
+      {"Stream_Switch_Slave_Mem_Trace_Slot2", 0x0003F3A8},
+      {"Stream_Switch_Slave_Mem_Trace_Slot3", 0x0003F3AC},
+      {"Stream_Switch_Event_Port_Selection_0", 0x0003FF00},
+      {"Stream_Switch_Event_Port_Selection_1", 0x0003FF04} };
+ 
+
+  std::string err;
+  std::string value;
+  dev->sysfs_get("aie_metadata", err, value);
+  if (!err.empty())
+    throw xrt_core::error(-EINVAL, err);
+  std::stringstream ss(value);
+  boost::property_tree::ptree pt;
+  boost::property_tree::read_json(ss, pt);
+   
+  int mKernelFD = open("/dev/dri/renderD128", O_RDWR);
+  if (!mKernelFD) {
+    throw xrt_core::error(-EINVAL, " Cannot open /dev/dri/renderD128");
+  }
+  XAie_DevInst* devInst;         // AIE Device Instance
+
+  XAie_SetupConfig(ConfigPtr,
+    pt.get<uint8_t>("aie_metadata.driver_config.hw_gen"),
+    pt.get<uint64_t>("aie_metadata.driver_config.base_address"),
+    pt.get<uint8_t>("aie_metadata.driver_config.column_shift"),
+    pt.get<uint8_t>("aie_metadata.driver_config.row_shift"),
+    pt.get<uint8_t>("aie_metadata.driver_config.num_columns"),
+    pt.get<uint8_t>("aie_metadata.driver_config.num_rows"),
+    pt.get<uint8_t>("aie_metadata.driver_config.shim_row"),
+    pt.get<uint8_t>("aie_metadata.driver_config.reserved_row_start"),
+    pt.get<uint8_t>("aie_metadata.driver_config.reserved_num_rows"),
+    pt.get<uint8_t>("aie_metadata.driver_config.aie_tile_row_start"),
+    pt.get<uint8_t>("aie_metadata.driver_config.aie_tile_num_rows"));
+
+  /* TODO get partition id and uid from XCLBIN or PDI, currently not supported*/
+  uint32_t partition_id = 1;
+  uint32_t uid = 0;
+  drm_zocl_aie_fd aiefd = { partition_id, uid, 0 };
+  int ret = ioctl(mKernelFD, DRM_IOCTL_ZOCL_AIE_FD, &aiefd) ? -errno : 0;
+  if (ret)
+    throw xrt_core::error(ret, "Create AIE failed. Can not get AIE fd Please load the xclbin");
+  int fd = aiefd.fd;
+
+  ConfigPtr.PartProp.Handle = fd;
+
+  AieRC rc;
+  XAie_InstDeclare(DevInst, &ConfigPtr);
+  if ((rc = XAie_CfgInitialize(&DevInst, &ConfigPtr)) != XAIE_OK)
+    throw xrt_core::error(-EINVAL, "Failed to initialize AIE configuration: " + std::to_string(rc));
+  devInst = &DevInst;
+  if(!devInst)
+    throw xrt_core::error(-EINVAL,"invalid aie object");
+
+  row+=1;
+  // TODO: get max row and column form aie_metadata
+  if(row <= 0 && row >= 9 && col < 0 && col >= 50)
```

Comment:
don't we already have it in aie_metadata? why not just use it?
And please correct the coding format (in some other places too), adding a space after key word. 
if ()

Reply:
done